cura/OAuth2/SecretStorage.py

```diff
@@ -0,0 +1,20 @@
+import keyring
+
+
+class SecretStorage:
+    def __init__(self):
+        self._stored_secrets = []
+
+    def __delitem__(self, key):
+        if key in self._stored_secrets:
+            del self._stored_secrets[key]
+            keyring.delete_password("cura", key)
+
+    def __setitem__(self, key, value):
+        self._stored_secrets.append(key)
+        keyring.set_password("cura", key, value)
+
+    def __getitem__(self, key):
```

Comment:
Missing typing

Reply:
You jumped the gun here @nallath ;-) Still very much in the "in progress" column at the moment.
Anyhow. I removed the SecretStorage class since I now Use a descriptor for the authenicationresponsemodel attributes